app/src/main/java/fr/free/nrw/commons/category/CategoryImagesActivity.java

```diff
@@ -102,17 +115,32 @@ public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
             FragmentManager supportFragmentManager = getSupportFragmentManager();
             supportFragmentManager
                     .beginTransaction()
-                    .replace(R.id.fragmentContainer, mediaDetails)
+                    .hide(supportFragmentManager.getFragments().get(supportFragmentManager.getBackStackEntryCount()))
+                    .add(R.id.fragmentContainer, mediaDetails)
```

Comment:
Aparently there is a reason you used "add" isdeat of "replace" here. Can you plase note this reason for further developers who will work on this code?